crates/apub/src/api/search.rs

```diff
@@ -55,49 +55,62 @@ pub async fn search(
   let creator_id = data.creator_id;
   let local_user = local_user_view.as_ref().map(|l| &l.local_user);
   let post_title_only = data.post_title_only;
+  let post_url_only = data.post_url_only;
+  let saved_only = data.saved_only;
+
+  let liked_only = data.liked_only;
+  let disliked_only = data.disliked_only;
```

Comment:
With how many variables that are just `let variable = data.variable` (13 by my count, we're using every single field of `Search`), this might be one of those cases where a destructure is less complicatied. e.g. on line 39:

```rust
  let Query(Search {
    q,
    community_id,
    community_name,
    creator_id,
    type_,
    sort,
    listing_type,
    page,
    limit,
    post_title_only,
    post_url_only,
    saved_only,
    liked_only,
    disliked_only
  }) = data;
```

Then, everywhere that follows the pattern I mentioned above will become redundant. Places like line 40 will just be like `let q = q.clone();` instead of `let q = data.q.clone();`.

Thoughts?

Reply:
Ya, I can do this, at least for the ones that don't need any manipulation.